pkg/sortition/sortition.go

```diff
@@ -64,53 +64,84 @@ func MonitorPool(
 func checkOperatorStatus(chain Chain) error {
```

Comment:
1 - yea, I didn't take that into account
2 - yes, I wasn't sure about it and that's why I created 2 commits with more and fewer logs when grouping the existing logic. Thx for clarification
3 - [This comment](https://github.com/keep-network/keep-core/pull/3085#discussion_r918839341) mentioned that `... if the pool is not locked, check if the eligibility can be restored.` That's why I thought that we cannot restore on the locked pool.

Anyway, pls look at this commit [e19ac0f](https://github.com/keep-network/keep-core/pull/3085/commits/e19ac0f18db1e83db9b32d75ad7d874b5ab118f9) I think all the cases are covered. If you don't like it, I can revert to the version that you suggested. I just wanted to simplify the existing logic.

Reply:
I am not sure if the new version simplifies the logic. True, the logic is more concise but I think following the execution path in if/else-if is harder than originally. For example this case: `else if !isPoolLocked` should be really read as: is not in the pool, is not up to date, and the pool is not locked. I think it's easier to reason about in the original version.

Reply:
Ok, I guess it's kind of subjective :) IMO when we enter `logger.Info("joining the sortition pool")` it also should be read as: operator is not up to date, is not in the pool, and the pool is not locked, but we only have `} else {` statement there. We implicitly cover this case and fall back here in both versions because other scenarios are handled at the top of the code. 

Here I reverted to the original code as you suggested: [8ce6a9e](https://github.com/keep-network/keep-core/pull/3085/commits/8ce6a9e46da425280afe04fb025829c68d66f475) & [79f86ea](https://github.com/keep-network/keep-core/pull/3085/commits/79f86ea2bd4a21a06c4f9b632e904f756403a535)